### PR TITLE
Copy Good kube config to nodes from first master

### PR DESCRIPTION
**Pull request by jurlwin**
# Proposed Changes

Fetch kube config from master node (using VIP) to /tmp/kube_config.<IP of Master>

Push that file to each agent node so that agent nodes can issue k3s commands and have a configuration. This is useful for patching, etc, when you have an ansible script and you want to drain a node, wait, reboot, check the node status, and uncordon for each patch

## Checklist

- [x ] Tested locally
- [x ] Ran `site.yml` playbook
- [x ] Ran `reset.yml` playbook
- [x ] Did not add any unnecessary changes
- [x ] 🚀

### CODE REVIEW

File: roles/k3s/master/tasks/main.yml [hunk 1]
```diff
@@ -153,13 +152,20 @@
     owner: "{{ ansible_user }}"
     mode: "u=rw,g=,o="
 
-- name: Configure kubectl cluster to https://{{ apiserver_endpoint | ansible.utils.ipwrap }}:6443
+- name: Configure kubectl cluster to https://{{ apiserver_endpoint }}:6443
   command: >-
     k3s kubectl config set-cluster default
-      --server=https://{{ apiserver_endpoint | ansible.utils.ipwrap }}:6443
+      --server=https://{{ apiserver_endpoint }}:6443
```
Comment (sleiner) [inline roles/k3s/master/tasks/main.yml:158]: Why did you remove this?

File: roles/k3s/master/tasks/main.yml [hunk 2]
```diff
@@ -64,8 +64,7 @@
     cmd: "systemd-run -p RestartSec=2 \
                       -p Restart=on-failure \
                       --unit=k3s-init \
-                      k3s server {{ server_init_args }} \
-                      {{ '--node-taint CriticalAddonsOnly=true:NoExecute' if k3s_single_node|bool == false else ''}}"
+                      k3s server {{ server_init_args }}"
```
Comment (sleiner) [inline roles/k3s/master/tasks/main.yml:67]: How is this related to your change?

File: roles/k3s/master/tasks/main.yml [hunk 3]
```diff
@@ -153,13 +152,20 @@
...
+      --server=https://{{ apiserver_endpoint }}:6443
       --kubeconfig ~{{ ansible_user }}/.kube/config
   changed_when: true
 
+- name: Fetch cluster config file from first master to push to agent nodes - allowing agent nodes to run k3s kubectl xxx (patching, convenience)
+  ansible.builtin.fetch:
+    src: ~{{ ansible_user }}/.kube/config
+    flat: true
+    dest: /tmp/kube_config.{{ apiserver_endpoint }}
+  when: ansible_hostname == hostvars[groups['master'][0]]['ansible_hostname']
```
Comment (sleiner) [inline roles/k3s/master/tasks/main.yml:167]: Could we get around this intermediate step by delegating the `copy` task for the `node` role to the first master?

File: roles/k3s/master/tasks/main.yml (hunk 1 above)
Comment (jurlwin) [reply roles/k3s/master/tasks/main.yml:158]: ok -- this is my mistake, 100%.  I thought I had the latest when I changed the file, but I didn't.  I need to redo this patch from the current.  This is totally me.  I worked on it, did what i needed, then copied the file in, instead of re-checking the master.  SMH.

File: roles/k3s/master/tasks/main.yml (hunk 2 above)
Comment (jurlwin) [reply roles/k3s/master/tasks/main.yml:67]: It's not, it's due to what I just sent related to my bad approach to this / my first pull request.  100% my fault.  But - frankly, -i'm not sure I want this for my use case, so I'm not sure If I want it applied to all machines.  my intent is to run longhorn on all nodes along with workloads as this is totally home-lab driven.  Shouldn't we let people put server args in the all.yaml in group_vars?  Maybe more of a comment there?

File: roles/k3s/master/tasks/main.yml (hunk 3 above)
Comment (jurlwin) [reply roles/k3s/master/tasks/main.yml:167]: technically, probably, yes.  Drawbacks to that -- at least as I see it.  I'm running ansible from my mac, across to 4 RPIs.  I didn't put my ansible user's ssh key in each node, rather, just from my mac.  I think this would be "harder" for users to get started by requiring ssh keys on the master node too...

Comment (jurlwin) [reply roles/k3s/master/tasks/main.yml:167]: oh, and I didn't delete the file, as I need it locally, eventually, anyway...  my post-process is to 'rename' the default user and default cluster name so that I can distinguish between two different clusters today.
